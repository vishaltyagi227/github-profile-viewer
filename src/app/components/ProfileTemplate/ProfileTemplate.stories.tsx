import React from 'react';
import { Story, Meta } from '@storybook/react';
import ProfileTemplate from '@components/ProfileTemplate';
import { IProfile } from '@containers/Profile/type';

export default {
  title: 'Components/ProfileTemplate',
  component: ProfileTemplate,
} as Meta;

const Template: Story<IProfile> = ({ ...args }) => (
  <ProfileTemplate {...args} />
);

export const Basic = Template.bind({});

Basic.args = {
  userDetails: {
    userName: 'vishalTyagi',
    followers: 0,
    following: 0,
    bio: 'FrontEnd Intern',
    email: null,
    location: '',
    url: '',
  },
};
