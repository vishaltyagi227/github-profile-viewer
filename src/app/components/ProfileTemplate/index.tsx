import React from 'react';
import {
  Card,
  CardContent,
  Avatar,
  Typography,
  CardActions,
} from '@mui/material';
import { Email, LocationOn, People } from '@mui/icons-material';

import {
  EMAIL_NOT_FOUND,
  LOCATION_NOT_FOUND,
  RELATION,
  TARGET,
  VARIANT,
} from '@constant/index';
import { IProfile } from '@containers/Profile/type';
import Link from '@components/Link';
import useStyles from './style';

const ProfileTemplate = ({ userDetails }: IProfile) => {
  const classes = useStyles();

  const {
    userName,
    avatarUrl,
    followers,
    following,
    bio,
    email,
    location,
    url,
    blog,
  } = userDetails;

  return (
    <Card className={classes.profileContent}>
      <div className={classes.avtarWrapper}>
        <Avatar alt={userName} src={avatarUrl} className={classes.avatar} />
      </div>
      <CardContent>
        <Typography
          gutterBottom
          variant={VARIANT.h4}
          className={classes.userName}
        >
          {userName}
        </Typography>
        <Typography variant={VARIANT.subtitle1} className={classes.bio}>
          {bio}
        </Typography>
        <div className={classes.infoWrapper}>
          <div>
            <Typography className={classes.info}>
              <People className={classes.icon} /> Followers {followers}
            </Typography>
            <Typography className={classes.info}>
              <People className={classes.icon} /> Following {following}
            </Typography>
          </div>
          <div>
            <Typography className={classes.info}>
              <Email className={classes.icon} />
              {email || EMAIL_NOT_FOUND}
            </Typography>
            <Typography className={classes.info}>
              <LocationOn className={classes.icon} />
              {location || LOCATION_NOT_FOUND}
            </Typography>
          </div>
        </div>
      </CardContent>
      <CardActions>
        <Link
          btn
          rel={RELATION.noopener}
          href={url}
          className={classes.profileBtn}
          target={TARGET.newTab}
        >
          Github Profile
        </Link>
        {blog && (
          <Link
            btn
            rel={RELATION.noopener}
            href={blog}
            className={classes.profileBtn}
            target={TARGET.newTab}
          >
            Blog
          </Link>
        )}
      </CardActions>
    </Card>
  );
};

export default ProfileTemplate;
