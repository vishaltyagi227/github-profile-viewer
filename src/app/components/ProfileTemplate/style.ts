import { createStyles, makeStyles } from '@mui/styles';
import { Theme } from '@mui/material/styles';

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    profileContent: {
      flex: '0 0 100%',
      maxWidth: theme.pxToRem(500),
      boxShadow: theme.shadows[2],
      padding: theme.spacing(5),
      [theme.breakpoints.down(theme.breakpoints.values.sm)]: {
        padding: theme.spacing(1),
      },
    },
    avtarWrapper: {
      display: 'flex',
      alignItems: 'center',
      justifyContent: 'center',
      paddingBlock: theme.spacing(2),
    },
    avatar: {
      width: theme.pxToRem(250),
      height: theme.pxToRem(250),
    },
    userName: {
      fontWeight: 500,
      textAlign: 'center',
      color: theme.customPallete.primary.main,
    },
    bio: {
      textAlign: 'center',
      color: theme.palette.text.secondary,
    },
    infoWrapper: {
      display: 'flex',
      alignItems: 'center',
      justifyContent: 'space-between',
      margin: `${theme.spacing(2)} auto`,
    },
    info: {
      display: 'flex',
      alignItems: 'center',
      '&': theme.typography.subtitle2,
      color: theme.palette.text.secondary,
      marginBottom: theme.spacing(1),
    },
    icon: {
      marginRight: theme.spacing(1),
      color: theme.palette.text.secondary,
    },
    profileBtn: {
      color: theme.customPallete.primary.main,
      backgroundColor: theme.customPallete.primary.light,
      '&:hover': {
        backgroundColor: theme.customPallete.secondary.light,
        color: theme.customPallete.secondary.main,
      },
      '& + &': {
        marginLeft: theme.spacing(2),
      },
    },
  })
);

export default useStyles;
