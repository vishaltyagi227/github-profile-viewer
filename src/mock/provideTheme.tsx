import React, { ReactElement } from 'react';
import { ThemeProvider } from '@mui/material/styles';
import theme from '@theme/index';

export default (ui: ReactElement): ReactElement => (
  <ThemeProvider theme={theme}>{ui}</ThemeProvider>
);
