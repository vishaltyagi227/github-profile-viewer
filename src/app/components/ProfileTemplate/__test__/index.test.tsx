import React from 'react';
import { cleanup, render, RenderResult } from '@testing-library/react';
import renderer from 'react-test-renderer';
import provideTheme from '@mock/provideTheme';
import ProfileTemplate from '../index';

const details = {
  userName: 'vishal',
  avatarUrl: 'https://dummyurl.com/img',
  followers: 0,
  following: 0,
  bio: 'Mock bio',
  email: null,
  location: 'Meerut',
  url: 'https://github.com/dummyvishal',
  blog: '',
};

describe('<ProfileTemplate />', () => {
  let container: RenderResult;

  afterEach(cleanup);

  it('Should render correctly', () => {
    container = render(provideTheme(<ProfileTemplate userDetails={details} />));
    expect(container.getByAltText(details.userName)).toBeInTheDocument();
    expect(
      container.getByRole('heading', { name: /vishal/i })
    ).toBeInTheDocument();
    expect(
      container.getByRole('link', { name: /github profile/i })
    ).toBeInTheDocument();
    expect(container.getByText(/following 0/i)).toBeInTheDocument();
    expect(container.getByText(/followers 0/i)).toBeInTheDocument();
    expect(container.getByTestId('EmailIcon')).toBeInTheDocument();
    expect(container.getByTestId('LocationOnIcon')).toBeInTheDocument();
  });
});

describe('<ProfileTemplate /> Snapshot', () => {
  afterEach(cleanup);

  it('Should match with Snapshot', () => {
    const tree = renderer
      .create(provideTheme(<ProfileTemplate userDetails={details} />))
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
