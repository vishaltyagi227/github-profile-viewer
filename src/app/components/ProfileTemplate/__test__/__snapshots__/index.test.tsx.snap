// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProfileTemplate /> Snapshot Should match with Snapshot 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root makeStyles-profileContent-10 css-bhp9pd-MuiPaper-root-MuiCard-root"
>
  <div
    className="makeStyles-avtarWrapper-11"
  >
    <div
      className="MuiAvatar-root MuiAvatar-circular makeStyles-avatar-12 css-1wlk0hk-MuiAvatar-root"
    >
      <img
        alt="vishal"
        className="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
        src="https://dummyurl.com/img"
      />
    </div>
  </div>
  <div
    className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
  >
    <h4
      className="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom makeStyles-userName-13 css-yeserc-MuiTypography-root"
    >
      vishal
    </h4>
    <h6
      className="MuiTypography-root MuiTypography-subtitle1 makeStyles-bio-14 css-10wpov9-MuiTypography-root"
    >
      Mock bio
    </h6>
    <div
      className="makeStyles-infoWrapper-15"
    >
      <div>
        <p
          className="MuiTypography-root MuiTypography-body1 makeStyles-info-16 css-ahj2mt-MuiTypography-root"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium makeStyles-icon-17 css-i4bv87-MuiSvgIcon-root"
            data-testid="PeopleIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16 11c1.66 0 2.99-1.34 2.99-3S17.66 5 16 5c-1.66 0-3 1.34-3 3s1.34 3 3 3zm-8 0c1.66 0 2.99-1.34 2.99-3S9.66 5 8 5C6.34 5 5 6.34 5 8s1.34 3 3 3zm0 2c-2.33 0-7 1.17-7 3.5V19h14v-2.5c0-2.33-4.67-3.5-7-3.5zm8 0c-.29 0-.62.02-.97.05 1.16.84 1.97 1.97 1.97 3.45V19h6v-2.5c0-2.33-4.67-3.5-7-3.5z"
            />
          </svg>
           Followers 
          0
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1 makeStyles-info-16 css-ahj2mt-MuiTypography-root"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium makeStyles-icon-17 css-i4bv87-MuiSvgIcon-root"
            data-testid="PeopleIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16 11c1.66 0 2.99-1.34 2.99-3S17.66 5 16 5c-1.66 0-3 1.34-3 3s1.34 3 3 3zm-8 0c1.66 0 2.99-1.34 2.99-3S9.66 5 8 5C6.34 5 5 6.34 5 8s1.34 3 3 3zm0 2c-2.33 0-7 1.17-7 3.5V19h14v-2.5c0-2.33-4.67-3.5-7-3.5zm8 0c-.29 0-.62.02-.97.05 1.16.84 1.97 1.97 1.97 3.45V19h6v-2.5c0-2.33-4.67-3.5-7-3.5z"
            />
          </svg>
           Following 
          0
        </p>
      </div>
      <div>
        <p
          className="MuiTypography-root MuiTypography-body1 makeStyles-info-16 css-ahj2mt-MuiTypography-root"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium makeStyles-icon-17 css-i4bv87-MuiSvgIcon-root"
            data-testid="EmailIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4-8 5-8-5V6l8 5 8-5v2z"
            />
          </svg>
          Email not found
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1 makeStyles-info-16 css-ahj2mt-MuiTypography-root"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium makeStyles-icon-17 css-i4bv87-MuiSvgIcon-root"
            data-testid="LocationOnIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C8.13 2 5 5.13 5 9c0 5.25 7 13 7 13s7-7.75 7-13c0-3.87-3.13-7-7-7zm0 9.5c-1.38 0-2.5-1.12-2.5-2.5s1.12-2.5 2.5-2.5 2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5z"
            />
          </svg>
          Meerut
        </p>
      </div>
    </div>
  </div>
  <div
    className="MuiCardActions-root MuiCardActions-spacing css-1t6e9jv-MuiCardActions-root"
  >
    <a
      className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root makeStyles-profileBtn-18 css-1e6y48t-MuiButtonBase-root-MuiButton-root"
      href="https://github.com/dummyvishal"
      onBlur={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      rel="noopener"
      tabIndex={0}
      target="_blank"
    >
      Github Profile
    </a>
    
  </div>
</div>
`;
